### DISCUSSION TITLE
Configutation in separate file

### DISCUSSION THREAD

**Discussion by pol22pol** [BlazejosP/huawei-sun2000-API-CLI#8] [Ideas]
Configuration in a separate file will allow for easier upgrade.

**Comment by BlazejosP** [comment]
Thanks for that idea that will be necessary to done one configuration file for both Kioskmode.sh&fusionsolarapp.sh which will include password username and configurations for influx.db and MQTT, Domoticz that are technologies which will be used by both files in not so long time. If you have your own proposition of file configuration structure  will be happy to see that. I 'm thinking also about writing additionally setup.sh which will use TUI for graphically configuration like in linux karnel compilation https://unix.stackexchange.com/questions/285777/how-do-i-create-a-terminal-based-gui/285778

This standard Linux TUI is called "Dialog"
I also thinking about use this TUI in standard files Kioskmode.sh&fusionsolarapp.sh but how? just to show on terminal data echo is all what we need  in most cases this may work in background and only insert data to influxDB so from this point of view the best resource usage is when show nothing at all on terminal.

**Comment by BlazejosP** [comment]
I add your proposition to milestones in issues

**Reply by BlazejosP** [reply]
> Configuration in a separate file will allow for easier upgrade.

Is already done

**Comment by BlazejosP** [comment]
Hello @pol22pol 
Just d'like to mention that in newest update config.conf is pushed to extreme can now work with many accounts and even with kioskmode links which workings without logins.

**Reply by pol22pol** [reply]
@BlazejosP, good job, thanks!
